craft_parts/overlays/layers.py

```diff
@@ -49,7 +51,8 @@ def for_part(cls, part: Part, *, previous_layer_hash: "LayerHash") -> "LayerHash
             to the given part.
         """
         hasher = hashlib.sha1()
-        hasher.update(previous_layer_hash.hash_bytes)
+        if previous_layer_hash:
+            hasher.update(previous_layer_hash.hash_bytes)
```

Comment:
The `hash_bytes` attribute in `LayerHash` is super-misleading. 

It looks like you're updating here with the **bytes** of the previous hash layer, but you're only using the **digest** of the previous hash layer. 

May this naming be improved? thanks!

Reply:
The name refers to the hash of the previous layer, expressed in bytes (as opposed to hexadecimal string). Since it can potentially confuse readers, I'm renaming to `digest`.